packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -1790,6 +1824,12 @@ class CollectDifferentUntilThresholdRound(CollectionRound, ABC):
     collection_key: str
     required_block_confirmations: int = 0
 
+    required_class_attributes: Tuple[str, ...] = (
+        "done_event",
+        "collection_key",
+        "required_block_confirmations",
```

Comment:
This isn't usually defined by the developer, is it?

Reply:
No, ideally if they are only inheriting it in a new class like `SomeNewRound(CollectDifferentUntilThresholdRound):` then they would just need to specify the attributes as usual, like `done_event = Event.DONE` etc.

Here, **we** set the `required_class_attributes` value internally, to check their present at runtime.